src/converter/modules/utils/dom-utils.js

```diff
@@ -175,48 +175,106 @@ export const createNewSectionForBlock = (document, block) => {
-  if (!main) return;
+  // Abort processing if a 'shade-box' is present to avoid modifying shaded sections.
+  const shadeBox = document.body.querySelector('div.shade-box');
+  if (shadeBox) return;
+
+  // Find the main content area; if it doesn't exist, stop further execution.
+  const mainContent = document.body.querySelector('main');
+  if (!mainContent) return;
+
+  // Try to find the first division within the main content, usually containing primary blocks.
+  const firstDiv = mainContent.querySelector('div');
+  if (!firstDiv) return;
+
   const sections = [];
   let currentSection = [];
+  let currentSectionLength = 0;
+
+  // Helper to finalize and reset the current section.
+  const finalizeCurrentSection = () => {
+    sections.push([...currentSection]);
+    currentSection = [];
+    currentSectionLength = 0;
+  };
+
+  // Helper to start a new section.
+  const createNewSection = (element, length) => {
+    finalizeCurrentSection();
+    currentSection.push(element);
+    currentSectionLength = length;
+  };
+
+  // Determine if an element is a header based on its tag level.
+  const isHeaderElement = (element) => getHeadingLevel(element) > -1;
 
-  const addToCurrentSection = (element, commit, isLast) => {
-    if (commit) {
-      if (currentSection.length) sections.push([...currentSection]);
-      currentSection = [element];
-    } else if (isLast) {
+  // Special case to handle continuous addition until a break condition is met.
+  const addRemainingElements = (initialElement, initialLength) => {
+    let element = initialElement;
+    let length = initialLength;
+    let continueAdding = true;
+    while (continueAdding) {
       currentSection.push(element);
-      sections.push([...currentSection]);
+      currentSectionLength += length;
+      if (element.tagName === 'div' || isHeaderElement(element)) {
+        continueAdding = false;
+      } else if (element.nextElementSibling) {
+        element = element.nextElementSibling.cloneNode(true);
+        length = element.textContent.length;
+      } else {
+        break;
+      }
+    }
+  };
+
+  // Function to decide whether to add elements to the current section or start a new one.
+  const addElementToSection = (element, isHeader, isLastElement) => {
+    const elementTextLength = element.textContent.length;
+
+    // Start a new section if adding the element exceeds the 1400 character limit.
+    if (currentSectionLength + elementTextLength > 1400) {
+      const lastElementIsHeader =
+        currentSection.length &&
+        isHeaderElement(currentSection[currentSection.length - 1]);
+
+      if (lastElementIsHeader && !isHeader) {
+        addRemainingElements(element, elementTextLength);
+      } else {
+        createNewSection(element, elementTextLength);
+      }
     } else {
+      // If the section limit is not exceeded, simply add the element.
       currentSection.push(element);
+      currentSectionLength += elementTextLength;
     }
```

Comment:
this is a bit overcomplicated..

All you need is:

Look at function: `createNewSectionForBlock`
 and create an equivalent that does  `createNewSectionForElements` and pass in the array of elements you want to make into a separate section.

Reply:
Created a new function createNewSectionForElements that processes each of the array of elements per section 
Temp Array  is currentSection which resets when the 1400 char limit is met